tests/io/iceberg/test_iceberg_writes.py

```diff
@@ -27,94 +55,123 @@ def local_catalog(tmpdir):
     return catalog
 
 
-def test_read_after_write_append(local_catalog):
+@pytest.fixture(
+    scope="function",
+    params=[
+        pytest.param(UNPARTITIONED_PARTITION_SPEC, id="unpartitioned"),
+        pytest.param(
+            PartitionSpec(PartitionField(source_id=1, field_id=1000, transform=IdentityTransform(), name="x")),
+            id="identity_partitioned",
+        ),
+        pytest.param(
+            PartitionSpec(PartitionField(source_id=1, field_id=1000, transform=BucketTransform(4), name="x")),
+            id="bucket_partitioned",
+        ),
+        pytest.param(
+            PartitionSpec(PartitionField(source_id=1, field_id=1000, transform=TruncateTransform(2), name="x")),
+            id="truncate_partitioned",
+        ),
+    ],
+)
+def simple_local_table(request, local_catalog):
+    schema = Schema(
+        NestedField(field_id=1, name="x", type=LongType()),
+    )
+
+    table = local_catalog.create_table("default.test", schema, partition_spec=request.param)
+    return table
+
+
+def test_read_after_write_append(simple_local_table):
     df = daft.from_pydict({"x": [1, 2, 3, 4, 5]})
     as_arrow = df.to_arrow()
-    table = local_catalog.create_table("default.test", as_arrow.schema)
-    result = df.write_iceberg(table)
+    result = df.write_iceberg(simple_local_table)
     as_dict = result.to_pydict()
-    assert as_dict["operation"] == ["ADD"]
-    assert as_dict["rows"] == [5]
-    read_back = daft.read_iceberg(table)
-    assert as_arrow == read_back.to_arrow()
+    assert all(op == "ADD" for op in as_dict["operation"]), as_dict["operation"]
+    assert sum(as_dict["rows"]) == 5, as_dict["rows"]
```

Comment:
Also, is there an assertion we should make here on the number of files that were written?

Reply:
Added